packages/backend/src/managers/gitManager.ts

```diff
@@ -33,4 +35,137 @@ export class GitManager {
       await simpleGit(gitCloneInfo.targetDirectory).checkout([gitCloneInfo.ref]);
     }
   }
+
+  async getRepositoryRemotes(directory: string): Promise<RemoteWithRefs[]> {
+    return simpleGit(directory).getRemotes(true);
+  }
+
+  async getRepositoryStatus(directory: string): Promise<StatusResult> {
+    return simpleGit(directory).status();
+  }
+
+  async pull(directory: string): Promise<void> {
+    const pullResult: PullResult = await simpleGit(directory).pull();
+    console.debug(`local git repository updated. ${pullResult.summary.changes} changes applied`);
+  }
+
+  async isGitInstalled(): Promise<boolean> {
+    try {
+      const version = await simpleGit().version();
+      return version.installed;
+    } catch (err: unknown) {
+      console.error(`Something went wrong while trying to access git: ${String(err)}`);
+      return false;
+    }
+  }
+
+  async processCheckout(gitCloneInfo: GitCloneInfo): Promise<void> {
+    // Check for existing cloned repository
+    if (existsSync(gitCloneInfo.targetDirectory) && statSync(gitCloneInfo.targetDirectory).isDirectory()) {
+      const result = await this.isRepositoryUpToDate(
+        gitCloneInfo.targetDirectory,
+        gitCloneInfo.repository,
+        gitCloneInfo.ref,
+      );
+
+      if (result.ok) {
+        return;
+      }
+
+      const error = `The repository "${gitCloneInfo.repository}" seems to already be cloned and is not matching the expected configuration: ${result.error}`;
+
+      // Ask user
+      const selected = await window.showWarningMessage(
+        `${error} By continuing, the AI application may not run as expected. `,
+        'Cancel',
+        'Continue',
+        result.updatable ? 'Update' : 'Reset',
+      );
+
+      if (selected === undefined || selected === 'Cancel') {
+        throw new Error('Cancelled');
+      }
+
+      if (selected === 'Continue') {
+        return;
+      }
+
+      if (selected === 'Update') {
+        // Update
+        await this.pull(gitCloneInfo.targetDirectory);
+        return;
+      }
+
+      if (selected === 'Reset') {
+        rmSync(gitCloneInfo.targetDirectory, { recursive: true });
+      }
```

Comment:
looks like we could avoid the return instructions using `if else if else if` pattern